File: pStudyWare20.Services/Implementations/NewsletterService.cs
using pStudyWare20.Repository.Interfaces;
using pStudyWare20.Services.Interfaces;
using pStudyWare20.Shared;

namespace pStudyWare20.Services.Implementations
{
    public class NewsletterService : INewsletterService
    {
        private readonly INewsletterRepository _newsletterRepository;

        public NewsletterService(INewsletterRepository newsletterRepository)
        {
            _newsletterRepository = newsletterRepository;
        }

        /// <summary>
        /// Saves newsletter subscription to AMC_tblNewsltr (legacy footer subscribe form).
        /// </summary>
        public NewsletterSubscribeResponse Subscribe(NewsletterSubscribeRequest request)
        {
            var response = new NewsletterSubscribeResponse();
            var email = request.Email?.Trim() ?? string.Empty;

            try
            {
                if (_newsletterRepository.EmailExistsAsync(email).GetAwaiter().GetResult())
                {
                    response.IsSuccess = true;
                    response.Message = "You are already subscribed to our newsletter.";
                    return response;
                }

                _newsletterRepository.AddSubscriptionAsync(email).GetAwaiter().GetResult();

                response.IsSuccess = true;
                response.Message = "Thank you for subscribing!";
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.ErrorMessage = ex.Message;
            }

            return response;
        }
    }
}

File: pStudyWare20.UI/src/services/newsletterService.js
import api from "./api";



class NewsletterService {

  async subscribe(email) {

    try {

      const response = await api.post("/Newsletter", {

        email: email.trim(),

      });

      return response.data;

    } catch (error) {

      const message =

        error.response?.data?.errorMessage ||

        error.response?.data?.message ||

        error.message ||

        "Unable to subscribe. Please try again.";

      throw new Error(message);

    }

  }

}



export default new NewsletterService();


